### PR TITLE
[ISSUE #4821] Add some integration tests for POP consumption in slave-acting-master mode

### LINKED ISSUE #4821: Add integration tests for POP consumption when slave is acting master
Slave acting master mode has been introduced in RIP-32, but there are few test cases to cover POP consumption mode when slave is acting master. It is necessary to add more test cases to guard this feature.

I'd like to add integration tests to check the behavior of POP consumption in slave-acting-master mode.

Test cases include:
3 broker containers are set up, each with 1 master and 2 slaves. In every case, N messages are sent to `Container1-Master1`.

|                   | consumed success (ACK)                                                   | reconsume later (non-ACK)                                                |
|-------------------|--------------------------------------------------------------------------|--------------------------------------------------------------------------|
| escaping locally  | - N messages should be consumed. <br>- No more messages can be consumed. | - N messages should be consumed. <br>- N retry messages can be consumed. |
| escaping remotely | - N messages should be consumed. <br>- No more messages can be consumed. | - N messages should be consumed. <br>- N retry messages can be consumed. |

### PR DESCRIPTION
**Pull request by caigy**
**Make sure set the target branch to `develop`**

## What is the purpose of the change
fix #4821 

## Brief changelog

Add `PopSlaveActingMasterIT` and fix some bugs found.

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: test/src/test/java/org/apache/rocketmq/test/container/PopSlaveActingMasterIT.java [hunk 1]
```diff
@@ -0,0 +1,569 @@
+package org.apache.rocketmq.test.container;
```
Comment (RongtongJin) [inline test/src/test/java/org/apache/rocketmq/test/container/PopSlaveActingMasterIT.java:1]: Missing ASF header

File: test/src/test/java/org/apache/rocketmq/test/container/PopSlaveActingMasterIT.java [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class PopSlaveActingMasterIT extends ContainerIntegrationTestBase {
```
Comment (RongtongJin) [inline test/src/test/java/org/apache/rocketmq/test/container/PopSlaveActingMasterIT.java:40]: How about ignoring these tests? It proves the function's correctness, but it takes too much time and is not the core function, which affects the running time of the CI. IMO, just run these tests manually before the release.

File: test/src/test/java/org/apache/rocketmq/test/container/PopSlaveActingMasterIT.java (hunk 1 above)
Comment (caigy) [reply test/src/test/java/org/apache/rocketmq/test/container/PopSlaveActingMasterIT.java:18]: Thanks, I'll add it.

File: test/src/test/java/org/apache/rocketmq/test/container/PopSlaveActingMasterIT.java (hunk 2 above)
Comment (caigy) [reply test/src/test/java/org/apache/rocketmq/test/container/PopSlaveActingMasterIT.java:59]: OK.

**Comment by ShannonDing** [comment]
org.apache.rocketmq.broker.processor.ChangeInvisibleTimeProcessorTest
testProcessRequest_Success(org.apache.rocketmq.broker.processor.ChangeInvisibleTimeProcessorTest)  Time elapsed: 2.568 sec  <<< ERROR!
java.lang.NullPointerException
	at org.apache.rocketmq.broker.processor.ChangeInvisibleTimeProcessorTest.testProcessRequest_Success(ChangeInvisibleTimeProcessorTest.java:122)

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**